torch/csrc/distributed/autograd/engine/dist_engine.h

```diff
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <gtest/gtest_prod.h>
```

Comment:
I don't like this... I know it's for making protected functions testable, but isn't there another way to achieve the same? For example, do the functions you're calling into need to be protected?